kernel/src/snapshot.rs

```diff
@@ -28,7 +24,6 @@ const LAST_CHECKPOINT_FILE_NAME: &str = "_last_checkpoint";
 pub struct Snapshot {
     pub(crate) table_root: Url,
     pub(crate) log_segment: LogSegment,
-    version: Version,
```

Comment:
I also wanted to flag this part. The version is just the `end_version` or effective version of the log segment. This seemed to me a property of `LogSegment` rather than a property of `Snapshot`. Let me know what you think :)

Reply:
yea if we do this can we have a `snapshot.version()` getter then? to just pass through the logic to ask it's log segment? I think in general we want to maintain the notion of a snapshot 'having a version'

EDIT you already have it :) this sounds good then!